### PR TITLE
Adapt kubernetes conformance test to new testsuite

### PR DESCRIPTION
**Pull request by ggkioulis**
Adapts the stack conformance test to the updated sonobuoy testsuite.

- Related ticket: https://progress.opensuse.org/issues/44303
- Needles: No new needles
- Verification run: http://skyrim.qam.suse.de/tests/5134

### CODE REVIEW

File: tests/caasp/stack_conformance.pm [hunk 1]
```diff
@@ -21,29 +21,48 @@ use testapi;
...
+my $config_json = <<'EOF';
+{
+  "plugins": [ { "name": "e2e" } ]
+}
+EOF
+
 sub run {
-    switch_to 'xterm';
+    # The repository for the sonobuoy package
+    my $repo = "https://download.opensuse.org/repositories/devel:/kubic/openSUSE_Tumbleweed/devel:kubic.repo";
```
Comment (mimi1vx) [inline tests/caasp/stack_conformance.pm]: ?  isn't better use `get_var('SOME_VAR_WITH_SONO_REPO')`

File: tests/caasp/stack_conformance.pm [hunk 2]
```diff
@@ -21,29 +21,48 @@ use testapi;
...
-    script_retry "kubectl -n sonobuoy logs sonobuoy | grep $sb_exit", retry => 90, delay => 60;
+    switch_to 'xterm';
+    become_root;
+
+    # Add the repo
+    zypper_call " ar -Gf $repo";
+
+    # Install the sonobuoy package on the controller node
+    zypper_call "-n in sonobuoy";
+
```
Comment (mimi1vx) [inline tests/caasp/stack_conformance.pm]: add disable/ remove sonobuy repo will be nice

File: tests/caasp/stack_conformance.pm (hunk 1 above)
Comment (ggkioulis) [reply tests/caasp/stack_conformance.pm]: thanks!!! Indeed it makes much more sense!

File: tests/caasp/stack_conformance.pm (hunk 2 above)
Comment (ggkioulis) [reply tests/caasp/stack_conformance.pm]: this conformance test is always the last one that runs, and the cluster is destroyed afterwards, so there will be no issues

**Comment by kravciak** [comment]
It's not compatible with CaaSP 2.0, which is still running in openQA:
 - https://openqa.suse.de/tests/2354701#step/stack_conformance/16

Is somebody reviewing CaaSP 2.0 / do we still support it / can we remove those tests?

**Comment by ggkioulis** [comment]
@kravciak  I ll check about CaaSP 2.0 on monday, when ppl will be back from vacation. If tests are still to be supported I ll revert to the e2e test of the older kubernetes version for CaaSP 2.

**Pull request merged**